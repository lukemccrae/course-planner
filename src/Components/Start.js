import React, {Component} from 'react';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Modal from 'react-modal';
import Countdown from 'react-countdown-now';
import Completionist from './Completionist';
import styled from 'styled-components';

const TimeBox = styled.div`
display: flex;
justify-content: space-between;
`


const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-40%',
    transform: 'translate(-50%, -50%)',
    width: '70%'
  }
};

class Start extends Component {
  constructor(props) {
    super(props)

    this.state = {
      modalIsOpen: false,
      promptModalIsOpen: false,
      currentTimerIndex: 0,
    }
    this.start = this.start.bind(this);
    this.openModal = this.openModal.bind(this);
    this.afterOpenModal = this.afterOpenModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.countdownDisplay = this.countdownDisplay.bind(this);
    this.nextTimer = this.nextTimer.bind(this);
    this.routineEnded = this.routineEnded.bind(this);
    this.formatCountdown = this.formatCountdown.bind(this);
    this.pause = this.pause.bind(this);
    this.unPause = this.unPause.bind(this);
  }



  openModal() {
    this.setState({modalIsOpen: true});
  }

  afterOpenModal() {
    // references are now sync'd and can be accessed.
    this.subtitle.style.color = 'black';
  }

  closeModal() {
    this.setState({
      modalIsOpen: false,
      currentTimerIndex: 0
    });
  }

  start() {
    this.openModal()
  }

  routineEnded() {
    if(this.state.currentTimerIndex < this.props.group.timers.length - 1) {
      return false;
    } else {
      return true;
    }
  }

  nextTimer() {
    if(this.routineEnded() === false) {
      let currentTimerIndex = this.state.currentTimerIndex;
      currentTimerIndex = ++currentTimerIndex;
      this.setState({currentTimerIndex}, () =>
      console.log()
      );
    } else {
      this.closeModal();
    }
  }

  formatCountdown(timer) {
    return Date.now() + timer.length * 1000;
  }

  pause() {
    console.log('pause');
    this.setState({
      paused: !this.state.paused
    })
  }

  unPause() {
    console.log('unpause');
    this.setState({
      paused: !this.state.paused
    })
  }

  countdownDisplay(timer) {
    let renderer = ({ minutes, seconds, completed }) => {
      //not sure if this will cause problems later.... if I leave seconds as a number it won't show two zeros
      if(minutes === 0) minutes = '00';
      if(seconds === 0) seconds = '00';
      if(seconds < 10 && seconds > 0) seconds = '0' + seconds;
      if (completed) {
        // Render a completed state
        return (<Completionist userId={this.props.userId} getTimers={this.props.getTimers} currentTimer={timer} nextTimer={this.nextTimer}></Completionist>)
      } else {
        // Render a countdown
        return (
          <div>
          <TimeBox>
            <span>{minutes}:{seconds}</span>
          </TimeBox>
          </div>
        );
      }
    };

      let countdownComponent = (
        <Countdown controlled={false} renderer={renderer} date={this.formatCountdown(timer)}>
          <Completionist getTimers={this.props.getTimers} currentTimer={timer} nextTimer={this.nextTimer}></Completionist>
        </Countdown>
      )

    let displayComponent = (
      <div>
        <p className="displayTime">{this.props.timeFormat(timer.length, 'str')}</p>
      </div>
    )

    let pausedComponent = (
      <div>
      <p className="displayTime">{this.props.timeFormat(timer.length, 'str')}</p>
    </div>
    )

    if(this.props.group.timers[this.state.currentTimerIndex] !== undefined) {
      if(timer.id === this.props.group.timers[this.state.currentTimerIndex].id) {
        return (
          countdownComponent
        )
      } else {
        return (
          displayComponent
        )
      }
    }
  }

  render() {
    return (
      <div>
        <Button onClick={this.start}>Start</Button>
        <Modal
          isOpen={this.state.modalIsOpen}
          onAfterOpen={this.afterOpenModal}
          onRequestClose={this.closeModal}
          style={customStyles}
          contentLabel="Example Modal"
          shouldCloseOnOverlayClick={false}
        >
        <div className="startNav">
          <button onClick={this.closeModal} type="button" className="close" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
          <Container>
            <h5 ref={subtitle => this.subtitle = subtitle}>{this.props.group.name}</h5>
            {this.props.group.timers.map(t => {
              return (
                <Row  className="countdownDisplay" key={t.id}>
                  <Col>
                    {t.name}
                  </Col>
                  <Col>
                    {this.countdownDisplay(t)}
                  </Col>
                </Row>
              )
            })}
          </Container>
        </div>
        </Modal>
      </div>
    )
  }

}

export default Start;
