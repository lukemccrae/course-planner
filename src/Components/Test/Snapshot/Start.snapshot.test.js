import React from 'react';
import Start from '../../Start.js';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });


const group = {
    name: 'group',
    timers: [{
      name: 'name',
      id: 1
    }]
  };

const timerFormat = jest.fn()

it('renders correctly', () => {
  const wrapper = shallow(<Start group={group} timeFormat={timerFormat} />);
  expect(wrapper).toMatchSnapshot();
});