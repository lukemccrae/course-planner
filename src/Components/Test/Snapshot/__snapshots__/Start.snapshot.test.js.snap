// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Start
    group={
      Object {
        "name": "group",
        "timers": Array [
          Object {
            "id": 1,
            "name": "name",
          },
        ],
      }
    }
    timeFormat={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
            "str",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef
          active={false}
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Start
        </ForwardRef>,
        <Modal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Example Modal"
          isOpen={false}
          onAfterOpen={[Function]}
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={false}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "bottom": "auto",
                "left": "50%",
                "marginRight": "-40%",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%, -50%)",
                "width": "70%",
              },
            }
          }
        >
          <div
            className="startNav"
          >
            <button
              aria-label="Close"
              className="close"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
            <ForwardRef
              fluid={false}
            >
              <ForwardRef
                className="countdownDisplay"
                noGutters={false}
              >
                <ForwardRef>
                  name
                </ForwardRef>
                <ForwardRef>
                  <Countdown
                    autoStart={true}
                    controlled={false}
                    date={NaN}
                    daysInHours={false}
                    intervalDelay={1000}
                    precision={0}
                    zeroPadTime={2}
                  >
                    <Completionist
                      nextTimer={[Function]}
                    />
                  </Countdown>
                </ForwardRef>
              </ForwardRef>
            </ForwardRef>
          </div>
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": "Start",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
          "variant": "primary",
        },
        "ref": null,
        "rendered": "Start",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "active": false,
            "disabled": false,
            "type": "button",
            "variant": "primary",
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ariaHideApp": true,
          "bodyOpenClassName": "ReactModal__Body--open",
          "children": <div
            className="startNav"
          >
            <button
              aria-label="Close"
              className="close"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
            <ForwardRef
              fluid={false}
            >
              <ForwardRef
                className="countdownDisplay"
                noGutters={false}
              >
                <ForwardRef>
                  name
                </ForwardRef>
                <ForwardRef>
                  <Countdown
                    autoStart={true}
                    controlled={false}
                    date={NaN}
                    daysInHours={false}
                    intervalDelay={1000}
                    precision={0}
                    zeroPadTime={2}
                  >
                    <Completionist
                      nextTimer={[Function]}
                    />
                  </Countdown>
                </ForwardRef>
              </ForwardRef>
            </ForwardRef>
          </div>,
          "closeTimeoutMS": 0,
          "contentLabel": "Example Modal",
          "isOpen": false,
          "onAfterOpen": [Function],
          "onRequestClose": [Function],
          "parentSelector": [Function],
          "portalClassName": "ReactModalPortal",
          "role": "dialog",
          "shouldCloseOnEsc": true,
          "shouldCloseOnOverlayClick": false,
          "shouldFocusAfterRender": true,
          "shouldReturnFocusAfterClose": true,
          "style": Object {
            "content": Object {
              "bottom": "auto",
              "left": "50%",
              "marginRight": "-40%",
              "right": "auto",
              "top": "50%",
              "transform": "translate(-50%, -50%)",
              "width": "70%",
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>,
              <ForwardRef
                fluid={false}
              >
                <ForwardRef
                  className="countdownDisplay"
                  noGutters={false}
                >
                  <ForwardRef>
                    name
                  </ForwardRef>
                  <ForwardRef>
                    <Countdown
                      autoStart={true}
                      controlled={false}
                      date={NaN}
                      daysInHours={false}
                      intervalDelay={1000}
                      precision={0}
                      zeroPadTime={2}
                    >
                      <Completionist
                        nextTimer={[Function]}
                      />
                    </Countdown>
                  </ForwardRef>
                </ForwardRef>
              </ForwardRef>,
            ],
            "className": "startNav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Close",
                "children": <span
                  aria-hidden="true"
                >
                  ×
                </span>,
                "className": "close",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": "×",
                },
                "ref": null,
                "rendered": "×",
                "type": "span",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef
                    className="countdownDisplay"
                    noGutters={false}
                  >
                    <ForwardRef>
                      name
                    </ForwardRef>
                    <ForwardRef>
                      <Countdown
                        autoStart={true}
                        controlled={false}
                        date={NaN}
                        daysInHours={false}
                        intervalDelay={1000}
                        precision={0}
                        zeroPadTime={2}
                      >
                        <Completionist
                          nextTimer={[Function]}
                        />
                      </Countdown>
                    </ForwardRef>
                  </ForwardRef>,
                ],
                "fluid": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <ForwardRef>
                        name
                      </ForwardRef>,
                      <ForwardRef>
                        <Countdown
                          autoStart={true}
                          controlled={false}
                          date={NaN}
                          daysInHours={false}
                          intervalDelay={1000}
                          precision={0}
                          zeroPadTime={2}
                        >
                          <Completionist
                            nextTimer={[Function]}
                          />
                        </Countdown>
                      </ForwardRef>,
                    ],
                    "className": "countdownDisplay",
                    "noGutters": false,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "name",
                      },
                      "ref": null,
                      "rendered": "name",
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Col",
                        "render": [Function],
                      },
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": <Countdown
                          autoStart={true}
                          controlled={false}
                          date={NaN}
                          daysInHours={false}
                          intervalDelay={1000}
                          precision={0}
                          zeroPadTime={2}
                        >
                          <Completionist
                            nextTimer={[Function]}
                          />
                        </Countdown>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "autoStart": true,
                          "children": <Completionist
                            nextTimer={[Function]}
                          />,
                          "controlled": false,
                          "date": NaN,
                          "daysInHours": false,
                          "intervalDelay": 1000,
                          "precision": 0,
                          "zeroPadTime": 2,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "nextTimer": [Function],
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": [Function],
                      },
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Col",
                        "render": [Function],
                      },
                    },
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "noGutters": false,
                    },
                    "render": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "fluid": false,
                },
                "displayName": "Container",
                "render": [Function],
              },
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Start
          </ForwardRef>,
          <Modal
            ariaHideApp={true}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            contentLabel="Example Modal"
            isOpen={false}
            onAfterOpen={[Function]}
            onRequestClose={[Function]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={false}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
            style={
              Object {
                "content": Object {
                  "bottom": "auto",
                  "left": "50%",
                  "marginRight": "-40%",
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%, -50%)",
                  "width": "70%",
                },
              }
            }
          >
            <div
              className="startNav"
            >
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
              <ForwardRef
                fluid={false}
              >
                <ForwardRef
                  className="countdownDisplay"
                  noGutters={false}
                >
                  <ForwardRef>
                    name
                  </ForwardRef>
                  <ForwardRef>
                    <Countdown
                      autoStart={true}
                      controlled={false}
                      date={NaN}
                      daysInHours={false}
                      intervalDelay={1000}
                      precision={0}
                      zeroPadTime={2}
                    >
                      <Completionist
                        nextTimer={[Function]}
                      />
                    </Countdown>
                  </ForwardRef>
                </ForwardRef>
              </ForwardRef>
            </div>
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": "Start",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
            "variant": "primary",
          },
          "ref": null,
          "rendered": "Start",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaHideApp": true,
            "bodyOpenClassName": "ReactModal__Body--open",
            "children": <div
              className="startNav"
            >
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
              <ForwardRef
                fluid={false}
              >
                <ForwardRef
                  className="countdownDisplay"
                  noGutters={false}
                >
                  <ForwardRef>
                    name
                  </ForwardRef>
                  <ForwardRef>
                    <Countdown
                      autoStart={true}
                      controlled={false}
                      date={NaN}
                      daysInHours={false}
                      intervalDelay={1000}
                      precision={0}
                      zeroPadTime={2}
                    >
                      <Completionist
                        nextTimer={[Function]}
                      />
                    </Countdown>
                  </ForwardRef>
                </ForwardRef>
              </ForwardRef>
            </div>,
            "closeTimeoutMS": 0,
            "contentLabel": "Example Modal",
            "isOpen": false,
            "onAfterOpen": [Function],
            "onRequestClose": [Function],
            "parentSelector": [Function],
            "portalClassName": "ReactModalPortal",
            "role": "dialog",
            "shouldCloseOnEsc": true,
            "shouldCloseOnOverlayClick": false,
            "shouldFocusAfterRender": true,
            "shouldReturnFocusAfterClose": true,
            "style": Object {
              "content": Object {
                "bottom": "auto",
                "left": "50%",
                "marginRight": "-40%",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%, -50%)",
                "width": "70%",
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  aria-label="Close"
                  className="close"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>,
                <ForwardRef
                  fluid={false}
                >
                  <ForwardRef
                    className="countdownDisplay"
                    noGutters={false}
                  >
                    <ForwardRef>
                      name
                    </ForwardRef>
                    <ForwardRef>
                      <Countdown
                        autoStart={true}
                        controlled={false}
                        date={NaN}
                        daysInHours={false}
                        intervalDelay={1000}
                        precision={0}
                        zeroPadTime={2}
                      >
                        <Completionist
                          nextTimer={[Function]}
                        />
                      </Countdown>
                    </ForwardRef>
                  </ForwardRef>
                </ForwardRef>,
              ],
              "className": "startNav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "Close",
                  "children": <span
                    aria-hidden="true"
                  >
                    ×
                  </span>,
                  "className": "close",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": "×",
                  },
                  "ref": null,
                  "rendered": "×",
                  "type": "span",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <ForwardRef
                      className="countdownDisplay"
                      noGutters={false}
                    >
                      <ForwardRef>
                        name
                      </ForwardRef>
                      <ForwardRef>
                        <Countdown
                          autoStart={true}
                          controlled={false}
                          date={NaN}
                          daysInHours={false}
                          intervalDelay={1000}
                          precision={0}
                          zeroPadTime={2}
                        >
                          <Completionist
                            nextTimer={[Function]}
                          />
                        </Countdown>
                      </ForwardRef>
                    </ForwardRef>,
                  ],
                  "fluid": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <ForwardRef>
                          name
                        </ForwardRef>,
                        <ForwardRef>
                          <Countdown
                            autoStart={true}
                            controlled={false}
                            date={NaN}
                            daysInHours={false}
                            intervalDelay={1000}
                            precision={0}
                            zeroPadTime={2}
                          >
                            <Completionist
                              nextTimer={[Function]}
                            />
                          </Countdown>
                        </ForwardRef>,
                      ],
                      "className": "countdownDisplay",
                      "noGutters": false,
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": "name",
                        },
                        "ref": null,
                        "rendered": "name",
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "Col",
                          "render": [Function],
                        },
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": <Countdown
                            autoStart={true}
                            controlled={false}
                            date={NaN}
                            daysInHours={false}
                            intervalDelay={1000}
                            precision={0}
                            zeroPadTime={2}
                          >
                            <Completionist
                              nextTimer={[Function]}
                            />
                          </Countdown>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "autoStart": true,
                            "children": <Completionist
                              nextTimer={[Function]}
                            />,
                            "controlled": false,
                            "date": NaN,
                            "daysInHours": false,
                            "intervalDelay": 1000,
                            "precision": 0,
                            "zeroPadTime": 2,
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {
                              "nextTimer": [Function],
                            },
                            "ref": null,
                            "rendered": null,
                            "type": [Function],
                          },
                          "type": [Function],
                        },
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "Col",
                          "render": [Function],
                        },
                      },
                    ],
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "defaultProps": Object {
                        "noGutters": false,
                      },
                      "render": [Function],
                    },
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "fluid": false,
                  },
                  "displayName": "Container",
                  "render": [Function],
                },
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
